MovingBlocks/Terasology
# Rendering engine: refactored 2 nodes + new node type

**[EDIT] the latest commit fixes the bug and makes this PR, further feedback notwithstanding, ready for merge.**

### Contains
* Refactored ShadowMapNode to fully take advantage of the new architecture
* Refactored WorldReflectionNode by extracting a new BackdropReflectionNode class from it, both taking advantage of the new architecture.
* [EDIT] Refactored BackdropNode to partially take advantage of the new architecture.
* Added and using a new BufferClearingNode class
* Added a new NodeFactory.createInstance(Class<T>, Object) method to allow passing initial data to the new Node.initialise(Object) method.
* [EDIT] <s>Changed FBOConfig class to hold information about the fbo manager to be associated with a given FBO.</s> - removed: will be reintroduced in a separate PR. 

### How to test

* Start Terasology
* Create/Load a game
* try changing Video Settings / Shadows
* try changing Video Settings / Water Reflections
* everything should look unchanged

### Outstanding before merging - [EDIT] OBSOLETE: the bug has been fixed.
At this stage this PR and in particular commit e737750e introduce a bug: when the Water Reflections are set to "Sky Only" the sky goes grey. This is probably due to the interactions between the old rendering engine architecture and the new one, as some nodes are still using the old architecture.

I intend to fix this bug before we merge this. However, most if not all the changes already visible in this PR are likely to be left untouched by the debugging process and I thought we could start the review process already.
